apps/allocations/app/store/allocation.js

```diff
@@ -101,13 +107,27 @@ const getAllocation = async ({ accountId, payoutId }) => {
         recurrences,
         token,
         date: new Date(startTime*1000),
-        recipients: { length: Number(recipientsLength) },
-        status: 2 //Approved will be made dynamic once the execution handler is integrated
+        recipients,
+        status: recipients.every(r => r.executions === recurrences) ? 3 : 2, //Approved will be made dynamic once the execution handler is integrated
```

Comment:
Can we remove this comment now?

Also, it would be great to assign these numbers to some constants, so we can give them meaningful names.